air/src/execution_step/value_types/stream/values_matrix.rs

```diff
@@ -18,6 +18,14 @@ use air_interpreter_data::GenerationIdx;
 
 use typed_index_collections::TiVec;
 
+fn get_stream_size_estimate<T>(values: &TiVec<GenerationIdx, Vec<T>>) -> usize {
+    use super::stream_definition::STREAM_MAX_SIZE;
+
+    values.iter().fold(0, |size, gen_values| {
```

Comment:
`try_fold` may give more elegant code here.

Reply:
I suggest adding a new field which represents values count in a matrix to not traverse it after adding every new value.

Reply:
Done